src/TextInput.js

```diff
@@ -25,14 +26,15 @@ const sizeVariants = variant({
   }
 })
 
-const TextInput = ({icon, className, ...rest}) => {
+const TextInput = ({icon, className, block, ...rest}) => {
+  // this class is necessary to style FilterSearch, plz no touchy!
   const wrapperClasses = classnames(className, 'TextInput-wrapper')
-  const inputClasses = classnames(icon ? 'input-icon' : 'input-no-icon')
-  const hasIcon = !!icon
+  const wrapperProps = pick(rest)
+  const inputProps = omit(rest)
```

Comment:
Neat 👍

Reply:
it's extremely useful for wrapper components like this!!